provider/pkg/gen/utilities.go

```diff
@@ -27,6 +31,53 @@ func apiNameToSdkName(name string) string {
 	return name
 }
 
+var nameRegex = regexp.MustCompile(`(?:(?:in[^\w]+(\w+(?:/[\w\{\}\*-\[\]]*)+)[^\w]+format))|(?:(?:Format|format|form|forms|formatted|pattern|Example|example|e\.g\.|such|Structured)(?: (?:of|is|as|like|will be))?[^\w]+(\w+(?:/[\w\{\}\*-\[\]]*)+))`)
```

Comment:
Would it be a little more readable to break the `or`ed regex up and apply separately?  Do we intend to capture all these groups or just the ones with the paths is sufficient? Could we use a named group for that, e.g. https://golang.org/pkg/regexp/#Regexp.SubexpNames.

Reply:
In general would be good to annotate regexes with an example or two of the type of strings they are supposed to match.

Reply:
I split the expression into two and added an example to each. Now there's just one subgroup that we care about, so it should be fine without a name.